### ISSUE TITLE
Inference Script for I2V model not producing 1:1 results compared to the validation videos (It feels worse somehow)

### ISSUE THREAD

**Issue by Yusixs** [a-r-r-o-w/cogvideox-factory#124]
### System Info / 系統信息

**Cuda:**
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0

**Diffusers**: 0.31.0                   pypi_0    pypi

**Python**: 3.10.15

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

1. Train a Lora using train_image_to_video_lora.sh.
2. Load saved lora and run an inference using the same validation image, prompt and seed. The result is somewhat similar but the results always feel worse compared to videos generated during validation.

Here's the bash script's parameters I used:

```bash
export TORCH_LOGS="+dynamo,recompiles,graph_breaks"
export TORCHDYNAMO_VERBOSE=1
export NCCL_P2P_DISABLE=1
export TORCH_NCCL_ENABLE_MONITORING=0
export CUDA_VISIBLE_DEVICES=1
export TOKENIZERS_PARALLELISM=false

GPU_IDS="3"

# Training Configurations
# Experiment with as many hyperparameters as you want!
LEARNING_RATES=("1e-4")
LR_SCHEDULES=("cosine_with_restarts")
OPTIMIZERS=("adamw")
MAX_TRAIN_STEPS=("3000")

# Single GPU uncompiled training
ACCELERATE_CONFIG_FILE="accelerate_configs/uncompiled_1.yaml"

# Absolute path to where the data is located. Make sure to have read the README for how to prepare data.
# This example assumes you downloaded an already prepared dataset from HF CLI as follows:
#   huggingface-cli download --repo-type dataset Wild-Heart/Disney-VideoGeneration-Dataset --local-dir /path/to/my/datasets/disney-dataset
DATA_ROOT="/data/users/sarim/data/cropped_video_dataset"
# DATA_ROOT="/data/users/sarim/PLLaVA/video_dataset_one"
CAPTION_COLUMN="prompts_new.txt"
VIDEO_COLUMN="videos_new.txt"




# Launch experiments with different hyperparameters
for learning_rate in "${LEARNING_RATES[@]}"; do
  for lr_schedule in "${LR_SCHEDULES[@]}"; do
    for optimizer in "${OPTIMIZERS[@]}"; do
      for steps in "${MAX_TRAIN_STEPS[@]}"; do
        output_dir="/data/users/sarim/cogvideox-factory/CHECKPOINTS/new_prompts_cogvideox-lora__optimizer_${optimizer}__steps_${steps}__lr-schedule_${lr_schedule}__learning-rate_${learning_rate}/"

        cmd="accelerate launch --config_file $ACCELERATE_CONFIG_FILE --gpu_ids $GPU_IDS training/cogvideox_image_to_video_lora.py \
          --pretrained_model_name_or_path THUDM/CogVideoX-5b-I2V \
          --data_root $DATA_ROOT \
          --caption_column $CAPTION_COLUMN \
          --video_column $VIDEO_COLUMN \
          --id_token BW_STYLE \
          --height_buckets 480 \
          --width_buckets 720 \
          --frame_buckets 49 \
          --dataloader_num_workers 8 \
          --pin_memory \
          --validation_prompt \"A driver stretching.\" \
          --validation_images \"/data/users/sarim/data/test_images/image0.jpg\"
          --validation_prompt_separator ::: \
          --num_validation_videos 3 \
          --validation_epochs 2 \
          --seed 42 \
          --rank 128 \
          --lora_alpha 128 \
          --mixed_precision bf16 \
          --output_dir $output_dir \
          --max_num_frames 49 \
          --train_batch_size 4 \
          --max_train_steps $steps \
          --checkpointing_steps 100 \
          --gradient_accumulation_steps 1 \
          --gradient_checkpointing \
          --learning_rate $learning_rate \
          --lr_scheduler $lr_schedule \
          --lr_warmup_steps 400 \
          --lr_num_cycles 1 \
          --enable_slicing \
          --enable_tiling \
          --noised_image_dropout 0.05 \
          --optimizer $optimizer \
          --beta1 0.9 \
          --beta2 0.95 \
          --weight_decay 0.001 \
          --max_grad_norm 1.0 \
          --allow_tf32 \
          --report_to wandb \
          --nccl_timeout 1800"
        
        echo "Running command: $cmd"
        eval $cmd
        echo -ne "-------------------- Finished executing script --------------------\n\n"
      done
    done
  done
done
```

Here's the inference python script I made. Please help me figure out if I'm missing something in the inference script which wasn't loaded or something :(

```python
import torch
from diffusers import CogVideoXImageToVideoPipeline, CogVideoXTransformer3DModel, CogVideoXDPMScheduler
from diffusers.utils import export_to_video, load_image
import os

# Set device
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

# Define paths and parameters
pretrained_model_name_or_path = "THUDM/CogVideoX-5b-I2V"  # Replace with your model path if local
lora_weights_path = "/data/users/sarim/cogvideox-factory/CHECKPOINTS/cogvideox-lora__optimizer_adamw__steps_3000__lr-schedule_cosine_with_restarts__learning-rate_1e-4" 
input_images_dir = "/data/users/sarim/data/test_images"  # Directory containing input images
output_dir = "/data/users/sarim/cogvideox-factory/InferenceResults_Seed42_DynamicCfgFalse"  
prompts = ["A driver stretching"]  # Prompt

guidance_scale = 6  
lora_rank = 128
lora_alpha = 128
load_dtype = torch.bfloat16
seed = 42

# Set random seed for reproducibility
generator = torch.Generator(device=device).manual_seed(seed) if seed else None

# Ensure the output directory exists
os.makedirs(output_dir, exist_ok=True)

# Load the transformer and scheduler
transformer = CogVideoXTransformer3DModel.from_pretrained(
    pretrained_model_name_or_path,
    subfolder="transformer",
    torch_dtype=load_dtype,
)

scheduler = CogVideoXDPMScheduler.from_pretrained(
    pretrained_model_name_or_path, 
    subfolder="scheduler"
)

# Initialize the pipeline
pipe = CogVideoXImageToVideoPipeline.from_pretrained(
    pretrained_model_name_or_path,
    transformer=transformer,
    scheduler=scheduler,
    torch_dtype=load_dtype,
)

pipe.vae.enable_slicing()
pipe.vae.enable_tiling()

# Load LoRA weights
pipe.load_lora_weights(lora_weights_path, adapter_name="cogvideox-lora")

# Apply LoRA scaling if needed 
pipe.set_adapters(["cogvideox-lora"], [lora_alpha / lora_rank ])

pipe = pipe.to(device)

model_config = transformer.module.config if hasattr(transformer, "module") else transformer.config

# Process each image in the input directory
input_images = [f for f in os.listdir(input_images_dir) if f.endswith(".jpg")]

for image_name in input_images:
    input_image_path = os.path.join(input_images_dir, image_name)

    # Load the input image
    input_image = load_image(input_image_path).resize((720, 480))

    for prompt in prompts:
        # Prepare sub-folder for the prompt
        prompt_folder_name = prompt.replace(" ", "_")
        prompt_output_dir = os.path.join(output_dir, prompt_folder_name)
        os.makedirs(prompt_output_dir, exist_ok=True)

        # Set up pipeline arguments
        pipeline_args = {
            "image": input_image,
            "prompt": prompt,
            "guidance_scale": guidance_scale,
            "use_dynamic_cfg": False,
            "max_sequence_length": model_config.max_text_seq_length,
            "num_frames": 49,
        }

        # Generate video
        video = pipe(**pipeline_args, generator=generator, output_type="np").frames[0]

        # Define the output video path
        video_name = f"{prompt_folder_name}_{os.path.splitext(image_name)[0]}.mp4"
        output_path = os.path.join(prompt_output_dir, video_name)

        # Export the video
        export_to_video(
            video, 
            output_path, 
            fps=8
        )

        print(f"Video saved to {output_path}")
```

### Expected behavior / 期待表现

I expected the video output to be almost identical (if not, very VERY similar to the validation video). However, it seems like I'm missing something in the inference script. Please help me out in identifying it if anyone has any idea ;-;

**Comment by Yusixs** [comment]
Update: I also included the set_seed() function from the accelerate.utils library but still no luck

**Comment by a-r-r-o-w** [comment]
The inference scripts looks good!

Have you tried a lower learning rate to see if it fixes quality problems? It might be the case that your model is getting worse due to the data being used, or something else causing it to get worse.

What are the results like when you use the normal model with images from your training dataset? 

With your lora, could you try setting the lora scale to something lower than 1.0 (currently it is `lora_alpha / lora_rank`, but you can set it to any value as such). At lower lora scale, do you see the outputs looking better visually? If yes, this probably means that your lora is making things worse because it could not capture the concept/style you're trying to train. At scale 1.0, do you see resemblance to the content you're trying to train with?

**Comment by sayakpaul** [comment]
Hey @Yusixs! I tried a full fine-tuning run on CogVideoX-5B and it seems to be going well: https://wandb.ai/sayakpaul/finetrainers-ltxv/runs/snv2aphi/overview. 

Did you have any luck w.r.t what you said in https://github.com/a-r-r-o-w/finetrainers/issues/124#issuecomment-2551791694?
